### PR TITLE
Avoid binding property signature in the process of creating symbols for accessors.

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review.

**Comment by 333fred** [comment]
Was this in response to a bug, or just something you noticed?

**Review by 333fred — APPROVED** [review]
LGTM (commit 1). You may have to merge in master to get the fixed build.

**Comment by AlekseyTs** [comment]
> Was this in response to a bug, or just something you noticed?

This is a step in the effort I mentioned offline to make sure we do not expose transient members that do not survive process of building members because other thread completed the work first.

**Comment by AlekseyTs** [comment]
> You may have to merge in master to get the fixed build.

It used to be that CI would merge with the current state of the target branch. Has this changed?

**Comment by 333fred** [comment]
> It used to be that CI would merge with the current state of the target branch. Has this changed?

That was my understanding as well, but your build does appear to be failing while master is not, so I'm now unsure.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off.

**Review by cston — APPROVED** [review]

File: src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs [hunk 1]
```diff
@@ -324,7 +206,7 @@ internal sealed override ImmutableArray<string> NotNullWhenFalseMembers
             }
 
             bool modifierErrors;
-            var declarationModifiers = this.MakeModifiers(modifiers, isExplicitInterfaceImplementation, hasBody || hasExpressionBody, location, diagnostics, out modifierErrors);
+            var declarationModifiers = this.MakeModifiers(modifiers, property.IsExplicitInterfaceImplementation, hasBody || hasExpressionBody, location, diagnostics, out modifierErrors);
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs:209]: >property.IsExplicitInterfaceImplementation [](start = 69, length = 42)

nit: consider extracting multiple uses to a local

**Comment by jcouv** [comment]
>             bool isExplicitInterfaceImplementation,

Not related to this PR but worth considering while in here:
Do we really need to pass `bool isExplicitInterfaceImplementation` in addition to `TypeSymbol? explicitInterfaceType` when constructing properties? Looking at the callers I think the `explicitInterfaceType` would be sufficient, and `isExplicitInterfaceImplementation` is redundant.

---
Refers to: src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs:91 in 9f41bc7. [](commit_id = 9f41bc7e112423683324dccab9521a16fd5f50e7, deletion_comment = False)

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 2)

**Comment by AlekseyTs** [comment]
>             bool isExplicitInterfaceImplementation,

> Looking at the callers I think the explicitInterfaceType would be sufficient, and isExplicitInterfaceImplementation is redundant.

If I remember correctly this isn't the case. Whether the property is an explicit implementation is a syntactic check, whereas explicitInterfaceType can be null if we failed to resolve the name.

---
In reply to: [770136693](https://github.com/dotnet/roslyn/pull/50898#issuecomment-770136693) [](ancestors = 770136693)

---
Refers to: src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs:91 in 9f41bc7. [](commit_id = 9f41bc7e112423683324dccab9521a16fd5f50e7, deletion_comment = False)

**Pull request merged**